schema/geometry_schema.json

```diff
@@ -839,25 +842,79 @@
 						"null"
 					]
 				},
+				"window_definition_type": {
+					"type": "string",
+					"enum": [
+						"Single Window",
+						"Multiple Windows",
+						"Window to Wall Ratio"
+					]
+				},
+				"wwr": {
+					"description": "Target window to wall ratio, only used if window_definition_type is \"Window to Wall Ratio\"",
+					"type": [
+						"number",
+						"null"
+					],
+					"minimum": 0,
+					"maximum": 1,
+					"exclusiveMaximum": true
+				},
 				"height": {
-					"description": "Vertical height of each window. In drawing units.",
-					"type": "number"
+					"description": "Vertical height of each window. In drawing units. Used if window_definition_type is \"Single Window\" or \"Multiple Windows\".",
+					"type": [
+						"number",
+						"null"
+					],
+					"minimum": 0,
+					"exclusiveMinimum": true
+				},
+				"window_spacing": {
+					"description": "Spacing between windows.  Only used if window_definition_type is  \"Multiple Windows\".",
+					"type": [
+						"number",
+						"null"
+					],
+					"minimum": 0,
+					"exclusiveMinimum": true
 				},
 				"width": {
-					"description": "Horizontal width of each window. In drawing units.",
-					"type": "number"
+					"description": "Horizontal width of each window. In drawing units. Used if window_definition_type is \"Single Window\" or \"Multiple Windows\".",
+					"type": [
+						"number",
+						"null"
+					],
+					"minimum": 0,
+					"exclusiveMinimum": true
 				},
 				"sill_height": {
 					"description": "Vertical distance from the floor to the window sill. In drawing units.",
-					"type": "number"
+					"type": [
+						"number",
+						"null"
+					],
+					"minimum": 0,
+					"exclusiveMinimum": true
+				},
+				"overhang_projection_factor": {
+					"description": "Projection factor for overhangs, set to null or 0 for no overhang.",
+					"type": [
+						"number",
+						"null"
+					],
+					"minimum": 0
```

Comment:
If this has the same meaning when it's null as when it's 0, I can just initialize it to 0 and then it doesn't need to permit `null`. Or is this something that only applies for some window definition types, and it would be confusing to have a value in some cases?